### PR TITLE
Replace Ballmer Peak example with Quadratic Formula

### PR DESCRIPTION
**Pull request by zpao**
This is a slightly more complicated example in the spirit of the Ballmer Peak
example, just without the spirits.

### CODE REVIEW

File: examples/quadratic/example.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  render: function() {
+    var a = this.state.a;
+    var b = this.state.b;
+    var c = this.state.c
+    var x1 = (-b + Math.sqrt(Math.pow(b, 2) - 4 * a * c)) / (2 * a);
+    var x2 = (-b - Math.sqrt(Math.pow(b, 2) - 4 * a * c)) / (2 * a);
+    return (
+      <div>
+        <strong>
+          <em>ax</em><sup>2</sup> + <em>bx</em> + <em>c</em> = 0
```
Comment (spicyj) [inline examples/quadratic/example.js]: needs more [KaTeX](http://khan.github.io/KaTeX/) ;)

File: examples/quadratic/example.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+      a: 1,
+      b: 3,
+      c: -4
+    };
+  },
+  handleChange: function(event) {
+    this.setState({
+      a: parseFloat(this.refs.a.getDOMNode().value),
+      b: parseFloat(this.refs.b.getDOMNode().value),
+      c: this.refs.c.getDOMNode().value
```
Comment (spicyj) [inline examples/quadratic/example.js]: why no parseFloat here?

File: examples/quadratic/example.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    var x2 = (-b - Math.sqrt(Math.pow(b, 2) - 4 * a * c)) / (2 * a);
+    return (
+      <div>
+        <strong>
+          <em>ax</em><sup>2</sup> + <em>bx</em> + <em>c</em> = 0
+        </strong>
+        <h4>Solve for <em>x</em>:</h4>
+        <p>
+          <label>
+            a: <input type="number" value={a} ref="a" onChange={this.handleChange} />
```
Comment (spicyj) [inline examples/quadratic/example.js]: I thought we were claiming

```
<input type="number" value={a} onChange={this.handleChange.bind(null, 'a')} />

setState: function(key, e) {
  var state = {};
  state[key] = parseFloat(e.target.value);
  this.setState(state);
}
```

was more idiomatic? this is okay with me too though.

File: examples/quadratic/example.js (hunk 2 above)
Comment (zpao) [inline examples/quadratic/example.js]: oops!

File: examples/quadratic/example.js (hunk 3 above)
Comment (zpao) [inline examples/quadratic/example.js]: I started writing that. Grabbing all the values at once felt a little bit easier to explain (even if we wouldn't normally do that). I actually kind of like it better too, we don't create 3 more functions on every render. `handle{A,B,C}Change` is another option but that's gross too.

**Comment by zpao** [comment]
Updated based on feedback.

**Pull request merged**